app/src/main/java/com/nytimes/android/sample/SampleApp.java

```diff
@@ -79,17 +76,13 @@ private void initPersister() {
                 .open();
     }
 
-    private BufferedSourceAdapter<RedditData> provideBufferedSourceAdapter() {
-        return new TwoWayParserAdapter<>(provideGson(), RedditData.class);
-    }
-
     private Persister<BufferedSource, BarCode> newPersister() throws IOException {
         return SourcePersisterFactory.create(getApplicationContext().getCacheDir());
     }
 
     private Observable<BufferedSource> fetcher(BarCode barCode) {
-        return provideRetrofit().fetchSubreddit(barCode.getKey(), "10").compose(new ObjectToSourceTransformer<>
```

Comment:
I'm not crazy about the syntax that makes you first create a TwoWayParserAdapter followed by a ObjectToSourceTransformer, that feels like an implementation detail.  How about we make a `SourceFromObjectConverterFactory` so that we can do `SourceFromObjectConverterFactory.create(gson,Type)` which can actually return the `Transformer`.  Feel free to play with the name if something sounds better than SourceFromObjectConverterFactory, Spring was my first love, thus my lack of being able to create short names :-)